packages/cardano-graphql-sps-provider/test/CardanoGraphQLStakePoolSearchProvider.test.ts

```diff
@@ -0,0 +1,50 @@
+import { CardanoGraphQLStakePoolSearchProvider } from '../src/CardanoGraphQLStakePoolSearchProvider';
+
+jest.mock('../src/fetchExtendedMetadata');
+jest.mock('../src/queryStakePoolsWithMetadata');
+
+const { fetchExtendedMetadata } = jest.requireMock('../src/fetchExtendedMetadata');
+const { queryStakePoolsWithMetadata } = jest.requireMock('../src/queryStakePoolsWithMetadata');
+
+describe('CardanoGraphQLStakePoolSearchProvider', () => {
+  const fragments = ['some', 'stake', 'pools'];
+  let provider: CardanoGraphQLStakePoolSearchProvider;
+
+  beforeEach(() => {
+    provider = new CardanoGraphQLStakePoolSearchProvider();
+  });
+
+  afterEach(() => {
+    queryStakePoolsWithMetadata.mockReset();
+    fetchExtendedMetadata.mockReset();
+  });
+
+  describe('queryStakePools', () => {
+    const stakePools = [{}, { metadata: { extDataUrl: 'some-url' } }, { metadata: {} }];
+    beforeEach(() => {
+      queryStakePoolsWithMetadata.mockResolvedValueOnce(stakePools);
+    });
+    describe('[fetchExt = false]', () => {
+      it('queries and returns stake pools without fetching extended metadata', async () => {
```

Comment:
We can probably remove `fetchExt` param and just treat it as the service's responsibility to maintain a local cache that includes it if available, for clients to naturally request alongside the other properties.

Reply:
> were you thinking to fetch it server-side

Yes, as it's just an extension of fetching the standard metadata from SMASH. We can make the service configurable to allow the operator control over this, but either way the client can be shielded quite easily here.